keps/0030-immutable-parameters.md

```diff
@@ -0,0 +1,173 @@
+---
+kep-number: 30
+short-desc: Immutable parameters that can only be set on installation
+title: Immutable parameters
+authors:
+  - @aneumann82
+owners:
+  - @aneumann82
+editor:
+creation-date: 2020-04-28
+last-updated: 2020-05-14
+status: provisional
+see-also:
+---
+
+## Summary
+
+This KEP describes a feature to allow parameters to be defined as immutable. Immutable parameters can not be updated after the installation of the operator
+
+## Table of Contents
+
+<!--ts-->
+      * [Summary](#summary)
+      * [Table of Contents](#table-of-contents)
+      * [Motivation](#motivation)
+         * [Goals](#goals)
+         * [Non-Goals](#non-goals)
+      * [Proposal](#proposal)
+         * [User Stories](#user-stories)
+         * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+            * [Installation](#installation)
+            * [Updates](#updates)
+            * [Upgrades](#upgrades)
+            * [Other](#other)
+         * [Risks and Mitigations](#risks-and-mitigations)
+      * [Alternatives](#alternatives)
+         * [Immutable Section in params.yaml](#immutable-section-in-paramsyaml)
+         * [A separate immutables.yaml](#a-separate-immutablesyaml)
+      * [Implementation History](#implementation-history)
+
+<!-- Added by: aneumann, at: Thu May 14 19:34:48 CEST 2020 -->
+
+<!--te-->
+
+## Motivation
+
+Certain aspects of an operator must be defined while installing an operator and can not be changed later on. 
+
+For example parameters that end up defining PVCs or the podManagementPolicy of a stateful set can not be changed after they are applied. Changing these parameters would lead to a broken state of the operator where the resources can not be applied again until the parameters are set back to their original value. In these cases an immutable parameter would prevent a user from accidentally break the operator.
+
+### Goals
+
+Make it possible for an operator developer to specify that a parameter can not be changed after the installation
+
+### Non-Goals
+
+- Allow modification of immutable parameters after the initial deploy plan is finished
```

Comment:
this is still confusing...  I don't know what this means... I would guess it is what you have below for the 3rd point.  but I'm not sure.    
if we are talking about modification of immutable params.. it is the goal to NOT allow that.. this is a strange statement which I think causes confusion.   The non-goals list below make sense.

Reply:
I've removed that